### PR TITLE
feat: Do not do VAPID auth on trusted connections

### LINKED ISSUE #938: Don't check Vapid sig value for partner endpoint users
Partners using cert'd endpoints have already validated themselves. 
Don't bother checking the signature block.

### PR DESCRIPTION
**Pull request by jrconlin**
Closes #938

### CODE REVIEW

File: autopush/web/webpush.py [hunk 1]
```diff
@@ -367,7 +367,11 @@ def validate_auth(self, d):
                 public_key = vapid_auth['k']
             else:
                 public_key = d["subscription"].get("public_key")
-            jwt = extract_jwt(token, public_key)
+            jwt = extract_jwt(
+                token,
+                public_key,
+                is_trusted=self.context['settings'].client_certs is not None
```
Comment (pjenvey) [inline autopush/web/webpush.py]: we have a settings.enable_tls_auth to use instead

File: autopush/jwt.py [hunk 2]
```diff
@@ -78,6 +78,20 @@ def extract_signature(auth):
         return payload, encoded
 
     @staticmethod
+    def extract(token):
```
Comment (bbangert) [inline autopush/jwt.py]: Maybe rename to ``extract_assertion`` ?

On first glance why you'd use decode vs. extract makes little sense without reading docs. Having slightly more info in the method name would help readability. Maybe they should both be ``decode_*`` for consistency, or ``extract_*``?

File: autopush/utils.py [hunk 3]
```diff
@@ -188,12 +188,14 @@ def decipher_public_key(key_data):
     raise ValueError("Unknown public key format specified")
 
 
-def extract_jwt(token, crypto_key):
-    # type: (str, str) -> Dict[str, str]
+def extract_jwt(token, crypto_key, is_trusted=False):
```
Comment (bbangert) [inline autopush/utils.py]: Maybe this new option is called ``verify``?

File: autopush/utils.py [hunk 4]
```diff
@@ -188,12 +188,14 @@ def decipher_public_key(key_data):
...
+    # type: (str, str, bool) -> Dict[str, str]
     """Extract the claims from the validated JWT. """
     # first split and convert the jwt.
     if not token or not crypto_key:
         return {}
+    if is_trusted:
+        return jwt.extract(token)
     return jwt.decode(token, decipher_public_key(crypto_key.encode('utf8')))
```
Comment (bbangert) [inline autopush/utils.py]: And maybe we either ``unverified_extract`` and ``verified_extract`` so its clear?

**Comment by bbangert** [comment]
Something about the method naming here feels prone to confusion, I think cleaning up the names slightly should alleviate that. Looks good other than that!

File: autopush/jwt.py (hunk 2 above)
Comment (jrconlin) [inline autopush/jwt.py]: Yeah, I was keeping with the original library which I am emulating, but no reason not to be clearer about things.

**Pull request merged**